internal/mempool/ids.go

```diff
@@ -30,6 +31,11 @@ func (ids *IDs) ReserveForPeer(peerID types.NodeID) {
 	ids.mtx.Lock()
 	defer ids.mtx.Unlock()
 
+	if removedID, ok := ids.peerMap[peerID]; ok && removedID > 0 {
```

Comment:
Does it matter what the ID is? Isn't the presence in the map sufficient for the reservation?
```suggestion
	if _, ok := ids.peerMap[peerID]; ok {
```